x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/PredictionFieldType.java

```diff
@@ -52,22 +52,25 @@ public Object transformPredictedValue(Double value, String stringRep) {
             case STRING:
                 return stringRep == null ? value.toString() : stringRep;
             case BOOLEAN:
-                if ((areClose(value, 1.0D) || areClose(value, 0.0D)) == false) {
-                    throw new IllegalArgumentException(
-                        "Cannot transform numbers other than 0.0 or 1.0 to boolean. Provided number [" + value + "]");
+                if (isNumberQuickCheck(stringRep)) {
+                    try {
+                        // 1 is true, 0 is false
+                        return Integer.parseInt(stringRep) == 1;
+                    } catch (NumberFormatException nfe) {
+                        // do nothing, allow fall through to final fromDouble
+                    }
+                } else if (isBoolQuickCheck(stringRep)) { // if we start with t/f case insensitive, it indicates boolean string
```

Comment:
There is one place where this might cause a mistake.

Lets assume the class labels are something that are NOT true or false, `taco` and `hamburger` for classes `0` and `1` respectively

If the user SPECIFICALLY selects the `boolean` prediction value type on inference, it will flag as follows:

`taco` will be `false` (since it starts with `t` but is not the word `true`)
`hamburger` will be `true` (since it doesn't start with `t` or `f`, we revert to the class number).

To me, this is acceptable, as the user would have to specify the `boolean` result type. Knowing their classes labels were NOT actually boolean :/.